common/src/Microsoft.Azure.IIoT.Validators.JsonSchemaDotNet/tests/Default/SchemaValidationTests.cs

```diff
@@ -0,0 +1,207 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace Microsoft.Azure.IIoT.Validators.JsonSchemaDotNet.Tests.Default {
+    using Microsoft.Azure.IIoT.Validators;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Text;
+    using System.Text.Json;
+    using Xunit;
+
+    /// <summary>
+    /// These tests are designed to exercise the generated schema for configuration file
+    /// validation. Tests can be additionally found in the schema generator tool located
+    /// at: https://github.com/WilliamBerryiii/opcpublisherschemavalidator
+    ///
+    /// The referenced schema file is a linked asset in the project file set to
+    /// copy to the output build directory, so that it can be easily referenced here.
+    /// </summary>
+    public class SchemaValidationTests {
+
+        // Allow training commas in the test parsers as the main serializer for
+        // this solution supports training commas.
+        private static JsonDocumentOptions parseOptions = new JsonDocumentOptions {
+            AllowTrailingCommas = true,
+        };
+
+        [Fact]
+        public void EnsureTestConfigurationPassesSchemaValidation() {
+            using var schemaReader = new StreamReader("Default/publishednodesschema.json");
+            var validator = new JsonSchemaDotNetSchemaValidator();
+            var results = validator.Validate(Encoding.UTF8.GetBytes(testConfiguration), schemaReader);
+            Assert.True(results.All(r => r.IsValid));
+        }
+
+        [Fact]
+        public void ByteStringNodeIdsThatAreNotByteStingsReturnErrors() {
+            using var schemaReader = new StreamReader("Default/publishednodesschema.json");
+
+            // Break the id of the first Node to ensure an error is thrown
+            var alteredConfig = testConfiguration.Replace("i=1001", "b=12345");
+            var validator = new JsonSchemaDotNetSchemaValidator();
+            var results = validator.Validate(Encoding.UTF8.GetBytes(alteredConfig), schemaReader);
+
+            // Wnsure that we failed the regex NodeID check
```

Comment:
`Wnsure` -> `Ensure`

Reply:
To be fixed.

Reply:
Typo changes will be pushed in next commit.